src/main/java/ZadaniaWzorceProjektowe_10_09/Zadanie4/WithdrawPower.java

```diff
@@ -0,0 +1,33 @@
+package ZadaniaWzorceProjektowe_10_09.Zadanie4;
+
+public abstract class WithdrawPower {
+    protected static final int BASE = 10;
+    protected WithdrawPower successor;
+
+    abstract protected int getDenomination();
+
+    abstract protected int getAllowable();
+
+    abstract protected String getRole();
+
+    public void setSuccessor(WithdrawPower successor) {
+        this.successor = successor;
+    }
+
+    public void processRequest(WithdrawRequest request) {
+        if (request.getAmount() <= this.getAllowable()) {
+            if (request.getAmount() / getDenomination() == 0) {
+                System.out.println("You will receive " + request.getAmount() + " PLN in " + this.getRole());
+            } else {
+                int integer = (request.getAmount() - ((request.getAmount() % getDenomination())));
+                System.out.println("You will receive " + integer + " PLN in " + this.getRole());
+                request.setAmount(request.getAmount() - integer);
+                successor.processRequest(request);
+            }
+        } else if ((successor != null)) {
+            request.setAmount(request.getAmount() - getAllowable());
```

Comment:
Wydaje mi się, że ten fragment kodu jest duplikowany:
            request.setAmount(request.getAmount() - getAllowable());
             System.out.println("You will receive " + getAllowable() + " PLN in " + this.getRole());
             successor.processRequest(request);

Może warto zrobić metodę która przyjmie jakąś kwotę i wykona te operacje. Musimy starać się unikać duplikacji w kodzie, co potem ułatwi nam fixowanie potencjalnych błędów.